service/frontend/workflowHandler.go

```diff
@@ -3022,6 +3168,46 @@ func (wh *WorkflowHandler) ListTaskListPartitions(ctx context.Context, request *
 	return resp, err
 }
 
+func (wh *WorkflowHandler) getRawHistory(
+	scope metrics.Scope,
+	domainID string,
+	execution gen.WorkflowExecution,
+	firstEventID int64,
+	nextEventID int64,
+	pageSize int32,
+	nextPageToken []byte,
+	transientDecision *gen.TransientDecisionInfo,
+	branchToken []byte,
+) ([]*gen.DataBlob, []byte, error) {
+	rawHistory := []*gen.DataBlob{}
+	shardID := common.WorkflowIDToHistoryShard(*execution.WorkflowId, wh.config.NumHistoryShards)
+
+	resp, err := wh.GetHistoryManager().ReadRawHistoryBranch(&persistence.ReadHistoryBranchRequest{
+		BranchToken:   branchToken,
+		MinEventID:    firstEventID,
+		MaxEventID:    nextEventID,
+		PageSize:      int(pageSize),
+		NextPageToken: nextPageToken,
+		ShardID:       common.IntPtr(shardID),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, data := range resp.HistoryEventBlobs {
+		encoding := gen.EncodingTypeThriftRW.Ptr()
+		if data.Encoding == common.EncodingTypeJSON {
+			encoding = gen.EncodingTypeJSON.Ptr()
+		}
```

Comment:
this is going to be error prone, e.g. if there will be a new encoding type, try
```go
switch data.Encoding {
    case common.EncodingTypeJSON:
        encoding = gen.EncodingTypeJSON.Ptr()
    case common.EncodingTypeThriftRW:
        encoding = gen.EncodingTypeThriftRW.Ptr()
    default:
        <panic or return a error>
```

Reply:
Done